src/cache.ts

```diff
@@ -22,23 +23,66 @@ export function memoryStorage() {
 
 const ONE_WEEK = 1000 * 60 * 60 * 24 * 7
 
-export function createAsyncStorage(storage: Storage) {
+interface CachedStorageOptions {
+  /**
+   * Namespace fragments to isolate cache keys.
+   *
+   * @example
+   * ```ts
+   * // Isolate cache per provider configuration
+   * const providerName = 'some-awesome-font-provider'
+   * const providerOptions = { apiKey: 'xxx', subset: 'latin' }
+   * createCachedAsyncStorage(storage, {
+   *   namespace: [providerName, providerOptions]
+   * })
+   * ```
+   */
+  namespace?: any[]
+}
+
+export function createCachedAsyncStorage(storage: Storage, options: CachedStorageOptions = {}) {
+  function resolveKey(key: string): string {
+    if (!options?.namespace || options.namespace.length === 0) {
+      return key
+    }
+
+    return `${createCacheKey(...options.namespace)}:${key}`
+  }
+
   return {
     async getItem<T = unknown>(key: string, init?: () => T | Promise<T>) {
+      const resolvedKey = resolveKey(key)
       const now = Date.now()
-      const res = await storage.getItem(key)
+      const res = await storage.getItem(resolvedKey)
       if (res && res.expires > now && res.version === version) {
         return res.data
       }
       if (!init) {
         return null
       }
       const data = await init()
-      await storage.setItem(key, { expires: now + ONE_WEEK, version, data })
+      await storage.setItem(resolvedKey, { expires: now + ONE_WEEK, version, data })
       return data
     },
     async setItem(key: string, data: unknown) {
-      await storage.setItem(key, { expires: Date.now() + ONE_WEEK, version, data })
+      await storage.setItem(resolveKey(key), { expires: Date.now() + ONE_WEEK, version, data })
     },
   }
 }
+
+function createCacheKey(...fragments: any[]): string {
+  const parts = fragments.map((f) => {
+    // No hash for string parts for better readability.
+    const part = typeof f === 'string' ? f : hash(f)
```

Comment:
For example, if we also hash values like the font provider’s name, it becomes difficult to identify files when checking what Storage has saved. So I think we shouldn’t hash string fields.